### DISCUSSION TITLE
passing the "directory of code location" where the code lives as a flag or config

### DISCUSSION THREAD

**Discussion by technophile-04** [vercel/vercel#10211] [Help]
### Problem Description

It would be really great if we could pass in the "directory of code location" where the code lives as an option 

Very naive eg : 
`vercel --code-location ./packages/nexts`

Reason : 
We have a starter kit/template which is `monorepo` having below directory stucture : 
```ts
root/
├── packages/
│   └── nextjs
|   |   ├─- vercel.json
|   └── hardhat
├── package.json
```

i.e we have two packages `nextjs` and `hardhat` we have `vercel` installed at the root of the project as mentioned in FAQ's of https://vercel.com/docs/concepts/monorepos

Now when you do `yarn vercel` at the root its asks you : 
![Screenshot 2023-04-20 at 9 10 13 PM](https://user-images.githubusercontent.com/80153681/233417569-2a84957d-3d32-41c5-bbaf-9a20157f2696.jpg)

It would really be great we could preconfigure it use `pacakges/nextjs` as the `source` where the code lives and users don't need specify the path themselves

Am I missing something ? Is this already built in verlce cli ? Is there a way to preconfigure it throught `vercel.json` ? If not I think this feature should be there built in vercel cli 😅 

checkout this comment -> https://github.com/scaffold-eth/scaffold-eth-2/pull/231#issuecomment-1471632149 for more context

Tysm 🙌🙏

**Comment by carletex** [comment]
+1

I think this will be great! Allowing passing config values on the CLI command.
